src/tests/pools/D3MPoolBase.t.sol

```diff
@@ -16,277 +16,106 @@
     }
 
-    function test_can_rely_deny() public {
-        assertEq(D3MPoolBase(d3mTestPool).wards(address(123)), 0);
+    function test_auth_modifier() public virtual {
+        WardsAbstract(address(pool)).deny(address(this));
 
-        D3MPoolBase(d3mTestPool).rely(address(123));
-
-        assertEq(D3MPoolBase(d3mTestPool).wards(address(123)), 1);
-
-        D3MPoolBase(d3mTestPool).deny(address(123));
-
-        assertEq(D3MPoolBase(d3mTestPool).wards(address(123)), 0);
+        checkModifier(address(pool), string(abi.encodePacked(contractName, "/not-authorized")), [
+            ID3MPool.quit.selector
+        ]);
     }
 
-    function test_cannot_rely_no_auth() public {
-        D3MPoolBase(d3mTestPool).deny(address(this));
-
-        assertRevert(address(d3mTestPool), abi.encodeWithSignature("rely(address)", address(123)), string(abi.encodePacked(contractName, "/not-authorized")));
+    function test_onlyHub_modifier() public virtual {
+        checkModifier(address(pool), string(abi.encodePacked(contractName, "/only-hub")), [
+            ID3MPool.deposit.selector,
+            ID3MPool.withdraw.selector,
+            ID3MPool.exit.selector
+        ]);
     }
 
-    function test_cannot_deny_no_auth() public {
-        D3MPoolBase(d3mTestPool).deny(address(this));
-
-        assertRevert(address(d3mTestPool), abi.encodeWithSignature("deny(address)", address(123)), string(abi.encodePacked(contractName, "/not-authorized")));
+    function test_quit_vat_caged() public virtual {
+        vat.cage();
+        vm.expectRevert(abi.encodePacked(contractName, "/no-quit-during-shutdown"));
+        pool.quit(address(this));
     }
 
-    function test_can_file_hub() public {
-        address newHub = address(new FakeHub(vat));
-        assertEq(CanLike(vat).can(d3mTestPool, hub), 1);
-        assertEq(CanLike(vat).can(d3mTestPool, newHub), 0);
-        D3MPoolBase(d3mTestPool).file("hub", newHub);
-        assertEq(CanLike(vat).can(d3mTestPool, hub), 0);
-        assertEq(CanLike(vat).can(d3mTestPool, newHub), 1);
-    }
+    function test_exit() public virtual {
+        TokenMock redeemableToken = TokenMock(pool.redeemable());
+        GodMode.setBalance(address(dai), address(pool), 1000 ether);
+        vm.prank(address(hub)); pool.deposit(1000 ether);
+        uint256 initialBalance = redeemableToken.balanceOf(address(pool));
+        end.setArt(100 ether);
 
-    function test_cannot_file_hub_no_auth() public {
-        D3MPoolBase(d3mTestPool).deny(address(this));
+        assertEq(redeemableToken.balanceOf(TEST_ADDRESS), 0);
+        assertEq(ExitLike(address(pool)).exited(), 0);
 
-        assertRevert(address(d3mTestPool), abi.encodeWithSignature("file(bytes32,address)", bytes32("hub"), address(123)), string(abi.encodePacked(contractName, "/not-authorized")));
-    }
+        vm.prank(address(hub)); pool.exit(TEST_ADDRESS, 10 ether);  // Exit 10%
 
-    function test_cannot_file_hub_vat_caged() public {
```

Comment:
I think this is one important to test at the base to make sure the implementation is safe for ES.

Reply:
Oops yes accidentally deleted this. Re-added.